pkg/adapter/configError.go

```diff
@@ -50,25 +50,27 @@ type ConfigError struct {
 // in configuration. The field parameter indicates the name of the
 // specific configuration field name that is problematic.
 func (e *ConfigErrors) Appendf(field, format string, args ...interface{}) *ConfigErrors {
-	if e == nil {
-		e = &ConfigErrors{}
+	ce := e
```

Comment:
Assigning to the variable 'e' has no impact on the caller of the method. Assigning to the receiver is generally bad form, like assigning to incoming method parameters.